server/src/termdb.gdc.status.js

```diff
@@ -0,0 +1,45 @@
+import path from 'path'
+import ky from 'ky'
+import serverconfig from '#src/serverconfig.js'
+
+/**
+ * Tests the GDC API status and returns true if the status is OK.
+ * If serverconfig.features.runRemainingWithoutAwait is false, it will throw an error if the status is not OK.
+ * @param ds
+ * @returns {Promise<boolean>}
+ */
+export async function testGDCApiStatus(ds) {
+	try {
+		const status = await getApiStatus(ds, 4)
+		if (status?.status !== 'OK') {
+			if (!serverconfig.features.runRemainingWithoutAwait) {
+				throw new Error('GDC API status is not OK')
+			} else {
+				return false
+			}
+		}
+	} catch (error) {
+		if (serverconfig.features.runRemainingWithoutAwait) {
+			return false
+		} else {
+			throw new Error('failed to get GDC API status: ' + (error.message || error))
+		}
+	}
+	return true
+}
+
+export async function getApiStatus(ds, retry = 0, retryDelay = 1000) {
+	const { host, headers } = ds.getHostHeaders()
+	try {
+		return await ky(path.join(host.rest, 'status'), { headers }).json()
```

Comment:
May also need to allow for non-json-encoded response, like error 404 in html. Native `fetch` may be able to catch this error and retry.